### PR TITLE
Fixes #20929: Revision are taken into account during generation

### PR DESCRIPTION
**Pull request by fanf**
https://issues.rudder.io/issues/20929

- correct the call to the "configurationRepository" that knows how to find object with revision
- correct the display of directive with revision (or technique with revision in the directive)
- correct the directive bundle sequence that should not have a forbidden `+` char

### CODE REVIEW

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/DeploymentService.scala [hunk 1]
```diff
@@ -725,7 +728,9 @@ trait PromiseGeneration_performeIO extends PromiseGenerationService {
 
   override def findDependantRules() : Box[Seq[Rule]] = roRuleRepo.getAll(true).toBox
   override def getAllNodeInfos(): Box[Map[NodeId, NodeInfo]] = nodeInfoService.getAll().toBox
-  override def getDirectiveLibrary(): Box[FullActiveTechniqueCategory] = roDirectiveRepository.getFullDirectiveLibrary().toBox
+  override def getDirectiveLibrary(ids: Set[DirectiveId]): Box[FullActiveTechniqueCategory] = {
+    configurationRepository.getDirectiveLibrary(ids).toBox
```
Comment (fanf) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/DeploymentService.scala:732]: this was the main problem

**Comment by fanf** [comment]
PR updated with a new commit

**Comment by fanf** [comment]
PR updated with a new commit

**Comment by fanf** [comment]
PR updated with a new commit

**Review by VinceMacBuche — APPROVED** [review]

**Review by VinceMacBuche — APPROVED** [review]

**Comment by VinceMacBuche** [comment]
OK, squash merging this PR

**Pull request merged**